File: src/api/incrementalReading/dto/dueIncrementalReadingDto.ts
import { IncrementalReadingPriority } from "../../cells/valueObjects/incrementalReading";

export default interface DueIncrementalReadingDto {
	cellId: string;
	fileId: string;
	title: string;
	priority: IncrementalReadingPriority;
	hasExtracts: boolean;
}

File: src-tauri/src/incremental_reading/dto/due_incremental_reading_dto.rs
use serde::Serialize;

use crate::{Guid, cells::value_objects::incremental_reading::IncrementalReadingPriority};

#[derive(Serialize)]
#[serde(rename_all = "camelCase")]
pub struct DueIncrementalReadingDto {
    pub cell_id: Guid,
    pub file_id: Guid,
    pub title: String,
    pub priority: IncrementalReadingPriority,
    pub has_extracts: bool,
}
